### ISSUE TITLE
net/haproxy Feature Request: FastCGI support

### ISSUE THREAD

**Issue by thuttinpasseron** [opnsense/plugins#2769]
**Is your feature request related to a problem? Please describe.**
Since HAProxy 2.1, it is possible to use HAProxy as a reverse proxy for FastCGI applications (see https://www.haproxy.com/fr/blog/load-balancing-php-fpm-with-haproxy-and-fastcgi/ and http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#10 for relevant examples and configuration options), for example PHP-FPM. Unfortunately this is not supported in the plugin, since one can't add a `fcgi-app` section to the configuration file from the web interface, or through pass-through options.

As was done in the NGINX plugin (namely #791), would it be possible to enable this function in the HAProxy plugin?

**Describe the solution you'd like**
I think that a new section could be added in the real servers dropdown menu for FastCGI apps with the most common options:

- Name (`fcgi-app` option)
- Path (`docroot` option)
- Index (`index` option)
- Path-info regex (`path-info` option)
- Add a pass-through field for advanced configuration

Also, in the Real Servers configuration screen, a dropdown menu could be added to select the protocol (`proto` option added after the server's IP and port) although this is not altogether necessary since the pass-through field already exists. A similar thing could be done in the Backend pool configuration screen for the selection of the `use-fcgi-app` option, again not altogether necessary since the passthrough field exists as well.

**Describe alternatives you've considered**
Another idea would consist in adding a pass-through config option to create new sections in the configuration file. Although this seems to go against how this plugin is developped as each pass-through field indicates that it shouldn't be used on a regular basis and that feature requests should be made in order to avoid having to use them.

Thanks in advance for reading this feature request.

**Comment by fraenki** [comment]
@thuttinpasseron Would you please provide an example configuration (haproxy.conf) that _you_ would like to use on OPNsense?

**Comment by thuttinpasseron** [comment]
Sure, here goes:

```
backend php1
  mode http
  use-fcgi-app php-fcgi-1            # Those can actually be added through passthrough fields
  server IP1 PORT1 proto fcgi        # in backends and real servers screens

backend php2
  mode http
  use-fcgi-app php-fcgi-2
  server IP2 PORT2 proto fcgi

fcgi-app php-fcgi-1
  log-stderr global
  docroot PATH1
  index index.php
  path-info ^(/.+\.php)(/.*)?$
  pass-header X-Forwarded-For
  pass-header X-Forwarded-Proto unless !ssl_fc
  pass-header X-Scheme unless !ssl_fc
  pass-header ...

fcgi-app php-fcgi-2
  log-stderr global
  docroot PATH2
  index filename.php
  path-info ^(/.+\.php)(/.*)?$

```
Hope this helps. Please feel free to ask if you need more information.

**Comment by fraenki** [comment]
This will be available in the upcoming os-haproxy version 3.12.

Some remarks:
* `pass-header` and `set-param` options are added by adding and linking a Rule
* the `proto fcgi` is automatically added to servers when a FastCGI application is enabled for a backend
* pass-through options are discouraged and not added, because AFAICT all FastCGI options are now supported
